python/feldera/pipeline.py

```diff
@@ -402,8 +406,23 @@ def query_tabular(self, query: str) -> Generator[str, None, None]:
         """
         Executes a SQL query on this pipeline and returns the result as a formatted string.
 
+        Important:
+            This method is lazy. It returns a generator and is not evaluated until you consume the result.
+
         :param query: The SQL query to be executed.
         :return: A generator that yields a string representing the query result in a human-readable, tabular format.
         """
 
         return self.client.query_as_text(self.name, query)
+
+    def execute(self, query: str, display: bool = True):
+        """
+        Executes an ad-hoc SQL query on this pipeline.
+
+        :param query: The SQL query to be executed.
+        :param display: True to print the query results in a human-readable tabular format. True by default.
+        """
+
+        resp = "\n".join(self.query_tabular(query))
```

Comment:
I don't think the `print` flag is necessary. If they want to output the result, they can use `query()` and print the result.
Also, it's unlikely to be an issue in practice, but this `"\n".join()` can be expensive if the query produces large output.  I think we should just consume the generator, dropping everything on the floor.